.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -0,0 +1,191 @@
+# Copyright The GOSST team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA Docker-based builder
+
+# TODO: permissions
+
+# TODO: env
+
+defaults:
+  run:
+    shell: bash
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    inputs:
+      builder-image:
+        description: >
+          The OCI image name of the builder image responsible for 
+          creating the artifact. This must not include a tag or digest.
+        required: true 
+        type: string
+      builder-digest:
+        description: >
+          The OCI image digest of the builder-image. 
+          This must be specified in order to fully specify the builder 
+          image. The image digest is of the form '<alg>:<digest>'
+        required: true 
+        type: string
+      config:
+        description: >
+          Path to a configuration file containing a command that the builder
+          image should be invoked with and optionally, docker options
+          and environment variables.
+        required: true 
+        type: string
+      output:
+        description: >
+          Path to the output artifact(s) relative to the root of the Git
+          repository.
+          Users may specify a single artifact or use a shell-style wildcard
+          pattern to match multiple artifacts.
+        required: true 
+        type: string
+      source-repository:
```

Comment:
do you think the source-repository and source-sha should be a single option, repo/name@sha1?
Just curious.

Reply:
I think Asra copied what I did in the container generator workflow for container images with the idea that it was clearer for 
folks to understand they *need* to use a digest rather than a tag, branch, etc. I don't have really strong opinions so I could go either way. It is maybe more natural for users the full `repo@sha`.

Reply:
In the case of the container generator at least it makes sense to split them up because actions like `docker/build-push-action` often return the image `digest` they built as a separate output.

Reply:
Yeah, the builder image and digest I'll definitely keep separate. Right now the buidlder binary in https://github.com/slsa-framework/slsa-github-generator/pull/1267 ingests source repo and sha separately. I could update this later though if we change both